OutOfSchool/OutOfSchool.WebApi/Services/IApplicationService.cs

```diff
@@ -85,7 +86,7 @@ public interface IApplicationService
     /// <param name="applicationDto">Application entity to update.</param>
     /// <param name="providerId">Id of the provider for workshop.</param>
     /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
-    Task<Result<ApplicationDto>> Update(ApplicationUpdate applicationDto, Guid providerId);
+    Task<Either<ErrorResponse, Result<ApplicationDto>>> Update(ApplicationUpdate applicationDto, Guid providerId);
```

Comment:
(Design issue) Type `Either<ErrorResponse, Result<ApplicationDto>` violates SRP principle in a way that you could describe failed state in 2 ways - either returning `Either.Left` (that is `ErrorResponse`) or returning `Either.Right` with failed `Result`. Please remove this discrepancy and fix the `Update` method implementation.

Reply:
Done